enhancements/microshift/microshift-default-cni.md

```diff
@@ -0,0 +1,290 @@
+necessary accelerators and I/O on-board, but are typically not
+extensible and are highly constrained on every resource, e.g.:
+
+* CPU: ARM Cortex-A35-class or Intel Atom class CPU, 2-4 cores, 1.5GHz clock
+* memory: 2-16GB RAM
+* storage: e.g. SATA3 @ 6Gb/s, 10kIOPS, 1ms (vs. NVMe @ 32Gb/s,
+  500kIOPS, 10µs)
+* network: Less bandwidth and less reliable than data center-based
+  servers, including being completely disconnected for extended
+  periods. Likely 1Gb/s NICs, potentially even LTE/4G connections at
+  5-10Mbps, instead of 10/25/40Gb/s NICs
+
+
+## Motivation
+
+MicroShift used flannel during the proof of concept phase, which later on
+was simplified to bridge-cni, being that pluggin the most minimalistic
+plugin which would provide Pod to Pod communication in a single-node environment
+and nothing else.
+
+A very important characteristic necessary for any CNI plugin used in MicroShift
+is a low-enough RAM, Disk, CPU footprint and boot times, as explained in the summary,
+the target platforms for MicroShift are small SBCs or SoCs where most of
+the computing capability must remain available for the final applications,
+otherwise MicroShift stops being useful in it's target environment.
+
+### User Stories
+
+An application developer builds applications where:
+1. Pods need to talk to each other via TCP o UDP connectivity.
+2. Pods need to talk to services on the LAN or Internet via TCP or UDP.
+3. Some pods need be exposed as NodePort types of services on the LAN.
+4. Some pods need to be exposed to other pods as ClusterIP services types.
+
+An application developer builds applications where some pods must
+have limited connectivity to other pods, or to the external network (i.e.
+plugin services from vendors which must only contact specific internal APIs)
+
+The edge device with MicroShift and the final application is deployed on an
+IPv6, or DualStack IPV4/IPV6 network.
+
+As a device owner I can protect the edge device with a host-level firewall
+(firewalld, iptables, nftables), and that type of configuration is compatible
+with the CNI.
+
+The edge device should be capable of deploying on a network with dynamic address
+provisioning like DHCP, SLAAC, or DHCPv6 and remain functional when the IP address
+changes.
+
+### Goals
+ 
+* Implement a default CNI that satisfies the described user histories while
+  minimizing the impact on footprint.
+
+* The CNI plugin should run without the ClusterNetworkOperator, providing
+  the `microshift-networking` elements any basic system configuration elements,
+  tools or dependencies to make the CNI plugin work.
+
+* Follow the [MicroShift design
+  principles](https://github.com/openshift/microshift/blob/main/docs/design.md)
+
+### Non-Goals
+
+* We do not plan to support multi-node clusters.
+* We are not trying to address high availability within the CNI, 
+  high availability should be acomplished by having multiple edge devices working
+  as single node MicroShifts.
```

Comment:
You could add a statement that while some of the optimizations made for OVN to work with MicroShift may also be applied to single-node OpenShift, that is out of scope for this enhancement and would need to be handled separately.